beacon-chain/state/state-native/getters_withdrawal.go

```diff
@@ -211,6 +174,54 @@ func (b *BeaconState) ExpectedWithdrawals() ([]*enginev1.Withdrawal, uint64, err
 	return withdrawals, processedPartialWithdrawalsCount, nil
 }
 
+func withdrawalTotal(ws []*enginev1.Withdrawal, idx primitives.ValidatorIndex) (total uint64) {
+	for _, w := range ws {
+		if w.ValidatorIndex == idx {
+			total += w.Amount
+		}
+	}
+	return
+}
+
+func (b *BeaconState) processPendingPartialWithdrawals(withdrawals []*enginev1.Withdrawal) ([]*enginev1.Withdrawal, uint64, uint64, error) {
+	withdrawalIndex := b.nextWithdrawalIndex
```

Comment:
this part is redefining already existing variables , this is safe for this function but must be considered for future changes